mbed cli 2: Rename command line options

jainvikas8 opened the pull request:
To ensure good usability with the Mbed CLI 2 the command line
options need to harmonize with the old Mbed CLI 1.

Therefore the following command options are changed by keeping the
existing functionality.

* `devices` to `detect`
* `init` to `new`
* `build` to `compile`
* `clone` to `import`

Further updated CMake minimum version requirement to 3.19.0

Recommend using BLE Advertising example instead of BLE LED

Added `deploy` command line option

jainvikas8 committed:
mbed cli 2: Add `deploy` command line option

Patater commented on docs/tools/mbed_cli_2/use.md:
```diff
@@ -13,7 +13,7 @@ For help:
 - To get help for a specific command, use `mbed-tools <command> --help`. For example, for helping with listing connected devices (the `devices` command):
```
```diff
- For example, for helping with listing connected devices (the `devices` command)
+ For example, for helping with listing connected devices (the `detect` command)
```

Patater commented on docs/tools/mbed_cli_2/use.md:
````diff
@@ -63,6 +63,22 @@ mbed-tools clone <example> <PATH>
+This ensures all dependencies are resolved and the versions are synchronised to the version specified in the library reference.
+
+    ```
+    mbedtools deploy <PATH>
````
Do we usually use `mbed-tools` or `mbedtools` in this document? Make everything consistent, without changing too many of them.

jainvikas8 committed:
mbed cli 2: Rename command line options

To ensure good usability with the Mbed CLI 2 the command line
options need to harmonize with the old Mbed CLI 1.

Therefore the following command options are changed by keeping the
existing functionality.

* `devices` to `detect`
* `init` to `new`
* `build` to `compile`
* `clone` to `import`

jainvikas8 committed:
mbed cli 2: Update CMake version requirements

Due to recent update of Mbed OS, CMake minimum version requirement
is 3.19.0.

Also recommend using BLE Advertising example instead of BLE LED.

jainvikas8 commented:
Forced pushed to address comments from @Patater

jainvikas8 commented:
Please do not merge before - https://github.com/ARMmbed/mbed-tools/pull/112

jainvikas8 commented:
New Mbed CLI 2 release pending - 3.6.0

Patater commented:
Mbed CLI 2 4.0.0 is up on pypi
